rmw_connext_shared_cpp/src/qos.cpp

```diff
@@ -85,3 +85,120 @@ get_datawriter_qos(
 
   return true;
 }
+
+template<typename DDSEntityQos>
+bool
+set_entity_qos_from_profile_generic(
+  const rmw_qos_profile_t & qos_profile,
+  DDSEntityQos & entity_qos)
+{
+  // Read properties from the rmw profile
+  switch (qos_profile.history) {
+    case RMW_QOS_POLICY_HISTORY_KEEP_LAST:
+      entity_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_HISTORY_KEEP_ALL:
+      entity_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT:
+      break;
+    default:
+      RMW_SET_ERROR_MSG("Unknown QoS history policy");
+      return false;
+  }
+
+  switch (qos_profile.reliability) {
+    case RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT:
+      entity_qos.reliability.kind = DDS::BEST_EFFORT_RELIABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_RELIABILITY_RELIABLE:
+      entity_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_RELIABILITY_SYSTEM_DEFAULT:
+      break;
+    default:
+      RMW_SET_ERROR_MSG("Unknown QoS reliability policy");
+      return false;
+  }
+
+  switch (qos_profile.durability) {
+    case RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL:
+      entity_qos.durability.kind = DDS::TRANSIENT_LOCAL_DURABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_DURABILITY_VOLATILE:
+      entity_qos.durability.kind = DDS::VOLATILE_DURABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT:
+      break;
+    default:
+      RMW_SET_ERROR_MSG("Unknown QoS durability policy");
+      return false;
+  }
+
+  if (qos_profile.depth != RMW_QOS_POLICY_DEPTH_SYSTEM_DEFAULT) {
+    entity_qos.history.depth = static_cast<DDS::Long>(qos_profile.depth);
+  }
+
+  if (qos_profile.deadline.sec != 0 || qos_profile.deadline.nsec != 0) {
```

Comment:
So if they both are 0 is this an invalid condition where we should be returning an error (or false as the case may be) or is this considered the "Default Value"? 

Applies to the below as well.

Reply:
0 is intended to mean "default value", yes - which is meant to mean infinity deadline or infinity lifespan

Reply:
https://community.rti.com/static/documentation/connext-dds/5.3.1/doc/api/connext_dds/api_cpp/structDDS__LifespanQosPolicy.html - if you look here those types have DDS_DURATION_INFINITE special value by default - didn't want to mess with setting that ourselved